OmsiHook/OmsiRemoteMethods.cs

```diff
@@ -1,27 +1,53 @@
 ﻿using System;
+using System.IO.Pipes;
 using System.Runtime.InteropServices;
+using OmsiHookRPCPlugin;
 
 namespace OmsiHook
 {
     /// <summary>
-    /// All of these methods will only work if called from a native Omsi plugin.
+    /// All of these methods will only work if called from a native Omsi plugin or if the OmsiHookRPCPlugin is installed.
     /// These methods also rely on OmsiHookInvoker.dll which must be in the Omsi plugins folder.
     /// </summary>
-	public class OmsiRemoteMethods : OmsiObject
+	public static class OmsiRemoteMethods
 	{
-        public OmsiRemoteMethods() : base() { }
+        private static NamedPipeClientStream pipe;
+        private static Memory memory;
 
-        internal OmsiRemoteMethods(Memory memory, int address) : base(memory, address) { }
+        public static bool IsInitialised => pipe?.IsConnected ?? false;
+
+        // TODO: Okay maybe this shouldn't be static... Singleton?
+        internal static void InitRemoteMethods(Memory omsiMemory, bool inifiniteTimeout = false)
+        {
+            memory = omsiMemory;
+
+#if !OMSI_PLUGIN
+            pipe = new(".", OmsiHookRPCMethods.PIPE_NAME, PipeDirection.InOut);
+            //pipe.ReadMode = PipeTransmissionMode.Message;
+            try
+            {
+                if (inifiniteTimeout)
+                    pipe.Connect();
+                else
+                    pipe.Connect(20000);
+            }
+            catch(TimeoutException)
+            {
+                pipe = null;
+                throw new TimeoutException("Couldn't manage to connect to OmsiHookRPCPlugin within 20 seconds! Check that it is loaded correctly.");
+            }
+#endif
+        }
 
         [Obsolete]
-        public int MakeVehicle()
+        public static int MakeVehicle()
         {
             int vehList = TTempRVListCreate(0x0074802C, 1);
             string path = @"Vehicles\GPM_MAN_LionsCity_M\MAN_A47.bus";
```

Comment:
I'm guessing this is demo data - probs shouldnt be pushed to prod, maybe bring to an example code or something 🤷